neuromation/api/plugins.py

```diff
@@ -0,0 +1,69 @@
+import numbers
+from typing import Any, Dict, List, Mapping, Tuple, Type, Union
+
+
+class ConfigBuilder:
+    _config_spec: Dict[str, Any]
+
+    def __init__(self) -> None:
+        self._config_spec = dict()
+
+    def _define_param(
+        self,
+        section: str,
+        name: str,
+        type: Union[
+            Type[bool],
+            Type[numbers.Real],
+            Type[numbers.Integral],
+            Type[str],
+            Tuple[Type[List[Any]], Type[bool]],
+            Tuple[Type[List[Any]], Type[str]],
+            Tuple[Type[List[Any]], Type[numbers.Real]],
+            Tuple[Type[List[Any]], Type[numbers.Integral]],
+        ],
+    ) -> None:
+        # Avoid circular import config.py <-> plugins.py
+        from .config import ConfigError
+
+        if section == "alias":
+            raise ConfigError("Registering aliases is not supported yet.")
+        if section in self._config_spec and name in self._config_spec[section]:
+            raise ConfigError(f"Config parameter {section}.{name} already registered")
+        self._config_spec.setdefault(section, dict())
+        self._config_spec[section][name] = type
+
+    def get_spec(self) -> Mapping[str, Any]:
```

Comment:
I doubt if the method should be public.
Maybe we will change the implementation completely.
`define_int()` and family has a great chance to survive as-as; `get_spec()` returns internal implementation details and very likely will be changed.

Also, please think about the documentation. How to describe the returned value to the end-user? Why the user need this method at all?